### PR TITLE
Set `usedforsecurity=False` in hashlib methods (FIPS compliance)

### PR DESCRIPTION
**Pull request by Wauplin**
Related to https://github.com/huggingface/transformers/issues/27034 and https://github.com/huggingface/huggingface_hub/pull/1782.

**TL;DR:** `hashlib` is not a secure library for cryptography-related stuff. We are only using `hashlib` for non-security-related purposes in `datasets` so it's fine. From Python 3.9 we set can `usedforsecurity=False` in any `hashlib` method which is mandatory for companies that forbid the use of `hashlib` for security purposes. This PR fixes that.

**Note:** before merging this we need to release a new tokenizers version that would allow the newest `huggingface_hub` version (see https://github.com/huggingface/tokenizers/pull/1385). Otherwise it might create friction to users that want to install `datasets` + `tokenizers` at the same time.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -64,7 +64,7 @@ jobs:
         run: pip install --upgrade pyarrow huggingface-hub dill
       - name: Install dependencies (minimum versions)
         if: ${{ matrix.deps_versions != 'deps-latest' }}
-        run: pip install pyarrow==8.0.0 huggingface-hub==0.18.0 transformers dill==0.3.1.1
+        run: pip install pyarrow==8.0.0 huggingface-hub==0.19.1 transformers dill==0.3.1.1
```
Comment (lhoestq) [inline .github/workflows/ci.yml:67]:
```suggestion
        run: pip install pyarrow==8.0.0 huggingface-hub==0.19.0 transformers dill==0.3.1.1
```

**Review by lhoestq — APPROVED** [review]
Feel free to merge after the releases

**Comment by Wauplin** [comment]
I'm finally merging as `transformers`/`tokenizers` dependency pins have been removed + `huggingface_hub 0.19.4` has fixed the deps incompatibility issue. All good now :)
